src/main/java/org/javarosa/xform/parse/FormInstanceParser.java

```diff
@@ -378,68 +386,105 @@ private void verifyItemsetSrcDstCompatibility (FormInstance instance) {
+                // groups, we need to register all descendant refs as targets for relevancy
+                // conditions to allow for chained reactions in triggerables registered in
+                // any of those descendants
+                for (TreeReference r : getDescendantRefs(instance, ec, ref))
+                    bind.relevancyCondition.addTarget(r);
+            }
+            if (bind.requiredCondition != null)
+                bind.requiredCondition.addTarget(ref);
+            if (bind.readonlyCondition != null)
+                bind.readonlyCondition.addTarget(ref);
+            if (bind.calculate != null)
+                bind.calculate.addTarget(ref);
+
+            // Initialize present nodes under the provided ref
+            for (TreeReference nodeRef : nodeRefs) {
+                TreeElement node = instance.resolveReference(nodeRef);
+                node.setDataType(bind.getDataType());
+
+                if (bind.relevancyCondition == null)
+                    node.setRelevant(bind.relevantAbsolute);
+                if (bind.requiredCondition == null)
+                    node.setRequired(bind.requiredAbsolute);
+                if (bind.readonlyCondition == null)
+                    node.setEnabled(!bind.readonlyAbsolute);
+                if (bind.constraint != null)
+                    node.setConstraint(new Constraint(bind.constraint, bind.constraintMessage));
+
+                node.setPreloadHandler(bind.getPreload());
+                node.setPreloadParams(bind.getPreloadParams());
+                node.setBindAttributes(bind.getAdditionalAttributes());
             }
         }
 
         applyControlProperties(instance);
     }
 
-    private static void attachBindGeneral (DataBinding bind) {
-        TreeReference ref = FormInstance.unpackReference(bind.getReference());
+    private Set<TreeReference> getDescendantRefs(FormInstance mainInstance, EvaluationContext evalContext, TreeReference original) {
+        // original has already been added to the 'toAdd' list.
 
-        if (bind.relevancyCondition != null) {
-            bind.relevancyCondition.addTarget(ref);
-        }
-        if (bind.requiredCondition != null) {
-            bind.requiredCondition.addTarget(ref);
-        }
-        if (bind.readonlyCondition != null) {
-            bind.readonlyCondition.addTarget(ref);
-        }
-        if (bind.calculate != null) {
-            bind.calculate.addTarget(ref);
+        Set<TreeReference> descendantRefs = new HashSet<>();
+        TreeElement repeatTemplate = mainInstance.getTemplatePath(original);
+        if (repeatTemplate != null) {
+            for (int i = 0; i < repeatTemplate.getNumChildren(); ++i) {
+                TreeElement child = repeatTemplate.getChildAt(i);
+                descendantRefs.add(child.getRef().genericize());
+                descendantRefs.addAll(getDescendantRefs(mainInstance, child));
+            }
+        } else {
```

Comment:
Do you want to add a todo about this branch being unreachable or do you think that since you're looking at that now you'll be doing something about it soon enough?